src/frontend/src/eth/components/loaders/LoaderMultipleEthTransactions.svelte

```diff
@@ -1,31 +1,56 @@
 <script lang="ts">
-	import { erc20UserTokensNotInitialized } from '$eth/derived/erc20.derived';
+	import { debounce, isNullish } from '@dfinity/utils';
+	import { ETHERSCAN_MAX_CALLS_PER_SECOND } from '$env/rest/etherscan.env';
 	import { enabledEthereumTokens } from '$eth/derived/tokens.derived';
 	import { loadEthereumTransactions } from '$eth/services/eth-transactions.services';
 	import { enabledErc20Tokens } from '$lib/derived/tokens.derived';
+	import { batch } from '$lib/services/batch.services';
 	import type { TokenId } from '$lib/types/token';
+	import type { ResultSuccess } from '$lib/types/utils';
 
 	// TODO: make it more functional
 	let tokensLoaded: TokenId[] = [];
 
 	const load = async () => {
-		if ($erc20UserTokensNotInitialized) {
+		if (isNullish($enabledEthereumTokens) || isNullish($enabledErc20Tokens)) {
 			return;
 		}
 
-		await Promise.allSettled(
-			[...$enabledEthereumTokens, ...$enabledErc20Tokens].map(
-				async ({ network: { id: networkId }, id: tokenId }) => {
-					if (!tokensLoaded.includes(tokenId)) {
-						await loadEthereumTransactions({ tokenId, networkId });
-						tokensLoaded.push(tokenId);
+		const promises = [...$enabledEthereumTokens, ...$enabledErc20Tokens].reduce<
```

Comment:
with the component, can you extract a function for this? it would make reading the `load` function and de facto the review a bit easier